### ISSUE TITLE
require-returns-check should have option to accept just one return (for Typescript)

### ISSUE THREAD

**Issue by noahtallen** [gajus/eslint-plugin-jsdoc#949]
## Expected behavior
The following function should not have any jsdoc errors:

```ts
/**
 * @param x A useful param.
 * @returns A useful string.
 */
function foo( x: boolean ): string | undefined {
	if ( x ) {
		return 'bar';
	}
}
```

It's typescript code, which means that types aren't encoded in the jsdoc `@returns` tag. You can solve this by putting `@returns {string|undefined}`, which isn't valuable (though it would be valuable for JS functions), or by adding `return` to the very end, which also isn't valuable.

Is this a bug or expected behavior? If it is expected behavior, I think the default rule shouldn't require empty `return` tags for this scenario!

While this can also happen for JS code, I think it's _somewhat_ useful, in that it forces you to put the correct type beside `@returns` if the function doesn't _always_ return something. But that's already covered by typescript.

## Actual behavior
This error occurs:
```
JSDoc @returns declaration present but return expression not available in function
```

## ESLint Config

```js
'jsdoc/require-returns-check': 'error'
```

## ESLint sample

See above

## Environment

- Node version: happens in both 16 and 18
- ESLint version: 8.14.0
- `eslint-plugin-jsdoc` version: 39.6.6 (latest)

**Comment by brettz9** [comment]
Rather than detecting "typescript" `mode` to enable this feature, I'm thinking we might just detect if a type is undefined, and in such a case, only check for one path to a `return`. We already have the `require-returns-type` rule to check for missing types, and I'd rather we keep typescript mode confined to the JSDoc flavor used by TypeScript rather than making assumptions such as that all TS users don't want to define types (one can check plain JS+JSDoc files using TS too).

**Comment by noahtallen** [comment]
That seems fair to me, since we'd be enforcing type information outside of JSDoc in this case anyways

**Comment by brettz9** [comment]
And sorry by "detect if a type is undefined", I should have said "detect if a type is absent". If a type is `undefined`, the same behavior would apply.
